From b3053274973af9d52d5cfd7b3f8fd2c3d2bebe0e Mon Sep 17 00:00:00 2001
From: William Escande <wescande@google.com>
Date: Wed, 26 Jun 2024 13:39:38 -0700
Subject: [PATCH 2/3] HeadsetService: Notify device changed at stop

Notify the active device changed to null when the HeadsetService stops.

Bug: 328847267
Fix: 328847267
Flag: Exempt shutdown cleanup
Test: None
Change-Id: Ia6ceee3b715f594bb057bac179cff61dcbd1231b
---
 .../app/src/com/android/bluetooth/hfp/HeadsetService.java    | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/android/app/src/com/android/bluetooth/hfp/HeadsetService.java b/android/app/src/com/android/bluetooth/hfp/HeadsetService.java
index d6354dd..37d0a45 100644
--- a/android/app/src/com/android/bluetooth/hfp/HeadsetService.java
+++ b/android/app/src/com/android/bluetooth/hfp/HeadsetService.java
@@ -225,7 +225,10 @@ public class HeadsetService extends ProfileService {
         unregisterReceiver(mHeadsetReceiver);
         synchronized (mStateMachines) {
             // Reset active device to null
-            mActiveDevice = null;
+            if (mActiveDevice != null) {
+                mActiveDevice = null;
+                broadcastActiveDevice(null);
+            }
             mInbandRingingRuntimeDisable = false;
             mForceScoAudio = false;
             mAudioRouteAllowed = true;
-- 
2.34.1

